### DISCUSSION TITLE
How to add additional non-pivot fields to attachment forms

### DISCUSSION THREAD

**Discussion by bkuhl** [laravel/nova-issues#5854] [Q&A]
When attaching a resource, I'd like to be able to add custom fields which do not actually belong on the resource but are used as a builder for the resource attachment's data. Take the below screenshot for example.  Depending on the "Dynamic Integration" selected I'd like to alter which fields are visible.  One might be a URL that we attempt to parse and extract data for the purpose of populating different fields..   

<img width="798" alt="Screenshot 2023-09-02 at 6 42 26 AM" src="https://github.com/laravel/nova-issues/assets/524933/5563e44e-5edf-4a5e-94b8-2fbea6a244a3">

Currently on "My Thing" in the screenshot I have this in the `fields()` relationship, however the notes field does not show up.

```
            BelongsToMany::make('Dynamic Integrations', 'integrations', \App\Nova\Integration::class)
                ->fields(function ($request, $relatedModel) {
                    return [
                        Text::make('Notes'),
                    ];
                }),
```

Is that I'm trying to do possible or am I missing something?

**Comment by bkuhl** [comment]
I think my better option at the moment to do what I need is to build a Standalone action for `integrations` so I can fully customize _how_ it's attached.

<img width="823" alt="Screenshot 2023-09-02 at 2 42 05 PM" src="https://github.com/laravel/nova-issues/assets/524933/3e067774-812e-4799-93de-f5d73edd11f6">

However in my case I didn't want this standalone action visible anywhere on the default index page.  So on the `Integration` resource I'm only registering it if it's an index page on the given resource where I want to use it:

```
class Integration extends Resource
{
    public function actions(NovaRequest $request)
    {
        // Because I'm dynamically hiding/showing fields based on UI selections, including the action for Action requests is important
        if ($request->viaResource() === Game::class || $request->isActionRequest()) {
            return [
                NewIntegration::make()->standalone()->onlyOnIndex(),
            ];
        }

        return [];
    }
```

**Comment by jdion84** [comment]
i think you want pivot fields: https://nova.laravel.com/docs/resources/relationships.html#pivot-fields

**Comment by bkuhl** [comment]
Well, I'm looking for a form field that isn't stored in the database, but I take that value and run some computations before saving maybe 3-4 other fields to the db based on that 1 value provided.

**Comment by jdion84** [comment]
just store them in the database bro

then you can use model observer to do what you want via a job or even just inside the observer methods themselves based on the value

it might be possible via a field `fillUsing` method though
